app/src/main/java/com/github/swent/echo/compose/components/HomeScreen.kt

```diff
@@ -0,0 +1,426 @@
+                                contentDescription = item.title
+                            )
+                        },
+                        badge = {
+                            item.badgeCount?.let { Text(text = item.badgeCount.toString()) }
+                        },
+                        modifier =
+                            Modifier.padding(NavigationDrawerItemDefaults.ItemPadding)
+                                .testTag("navigation_item_$index")
+                    )
+                }
+            }
+        },
+        drawerState = drawerState,
+        modifier = Modifier.testTag("hamburger_menu"),
+        gesturesEnabled = false
+    ) {
+        Scaffold(
+            modifier = Modifier.testTag("home_screen"),
+            topBar = {
+                CenterAlignedTopAppBar(
+                    colors =
+                        TopAppBarDefaults.topAppBarColors(
+                            containerColor = MaterialTheme.colorScheme.primaryContainer,
+                            titleContentColor = MaterialTheme.colorScheme.onPrimary,
+                        ),
+                    title = {
+                        Text(
+                            "Echo",
+                            maxLines = 1,
+                            overflow = TextOverflow.Ellipsis, // should not happen
+                            fontWeight = FontWeight.SemiBold
+                        )
+                    },
+                    navigationIcon = { // hamburger menu
+                        IconButton(
+                            onClick = { scope.launch { drawerState.open() } },
+                            modifier = Modifier.testTag("menu_button")
+                        ) {
+                            Icon(
+                                imageVector = Icons.Filled.Menu,
+                                contentDescription = "Menu to access all subpages"
+                            )
+                        }
+                    },
+                    actions = { // search icon
+                        IconButton(
+                            onClick = {
+                                if (mode.value == MapOrListMode.MAP) {
+                                    mode.value = MapOrListMode.LIST
+                                } else {
+                                    mode.value = MapOrListMode.MAP
+                                }
+                            },
+                            modifier = Modifier.testTag("list_map_mode_button")
+                        ) {
+                            Icon(
+                                imageVector = Icons.Filled.List,
+                                contentDescription = "Search icon to access the search screen"
+                            )
+                        }
+                    },
+                    scrollBehavior = scrollBehavior,
+                    modifier = Modifier.testTag("top_bar")
+                )
+            },
```

Comment:
Can't we extract this into the TopBar.kt file ?

Reply:
out of scope for now; askip @ntissieres tried to do this but it somehow didn't work... 
I'll create an issue for this...